lib/src/presentation/bloc/movie_bloc.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:async';
+
+import '../../core/bloc/i_bloc.dart';
+import '../../core/usecase/i_usecase.dart';
+import '../../core/util/data_state.dart';
+import '../../domain/entity/cast.dart';
+import '../../domain/entity/event.dart';
+import '../../domain/entity/movie.dart';
+import '../../domain/usecase/fetch_first_last_movie_usecase.dart';
+import '../../domain/usecase/load_cast_to_movie_usecase.dart';
+
+class MovieBloc extends IBloc {
+  final StreamController<Event<Movie>> _movieStreamController =
+      StreamController<Event<Movie>>.broadcast();
+
+  final StreamController<Event<List<Cast>>> castStreamController =
+      StreamController<Event<List<Cast>>>.broadcast();
+
+  final IUseCase<Movie, dynamic> _fetchFirstMovie;
+  final IUseCase<Future<DataState<List<Cast>>>, Movie> _loadCast;
+
+  MovieBloc({
+    IUseCase<Movie, dynamic>? fetchFirstMovie,
+    IUseCase<Future<DataState<List<Cast>>>, Movie>? loadCast,
+  })  : _fetchFirstMovie = fetchFirstMovie ?? FetchFirstLastMovieUseCase(),
+        _loadCast = loadCast ?? LoadCastToMovieUseCase();
+
+  Movie getMockMovie() {
+    return _fetchFirstMovie.call();
+  }
+
+  void loadCast(Movie movie) async {
+    DataState<List<Cast>> cast = await _loadCast.call(movie);
```

Comment:
Move the logic to the cast to an specific bloc, like CastBloc

Reply:
Done